x-pack/plugins/global_search_bar/public/components/search_bar.tsx

```diff
@@ -187,26 +182,23 @@ export const SearchBar: FC<SearchBarProps> = (opts) => {
           types: rawParams.filters.types,
           tags: tagIds,
         };
-        // TODO technically a subtle bug here
-        // this term won't be set until the next time the debounce is fired
-        // so the SearchOption won't highlight anything if only one call is fired
-        // in practice, this is hard to spot, unlikely to happen, and is a negligible issue
-        setSearchTerm(rawParams.term ?? '');
```

Comment:
I'm not seeing a huge reason for `searchTerm`/`setSearchTerm` to exist. See https://github.com/elastic/kibana/pull/85084/files#diff-8ea278ca607222ad8a87a3dcd86411eecda56798f8cb87e158aeca1c80cc4f71R388